### PR TITLE
Add "Jump emitter forward in time on creation" property

### PR DESCRIPTION
**Pull request by tristanbob**
Following these instructions from @4ian: https://github.com/4ian/GDevelop/pull/4985#issuecomment-1443120194

### CODE REVIEW

**Comment by tristanbob** [comment]
The property seems to be working properly (the editor looks good and it saves changes to the value).

However, the emitter is not jumping forward in time.  

Did I call the update() function correctly?

Did I put it in the right location of the constructor?

File: Extensions/ParticleSystem/particleemitterobject.ts [hunk 1]
```diff
@@ -158,8 +160,13 @@ namespace gdjs {
       this.particleRotationMaxSpeed = particleObjectData.particleAngle2;
       this.maxParticlesCount = particleObjectData.maxParticleNb;
       this.additiveRendering = particleObjectData.additive;
+      this.jumpForwardInTimeOnCreation =
+        particleObjectData.jumpForwardInTimeOnCreation;
       this._textureDirty = this.texture !== '';
 
+      // Jump emitter forward in time
+      this._renderer.update(this.jumpForwardInTimeOnCreation);
```
Comment (4ian) [inline Extensions/ParticleSystem/particleemitterobject.ts:168]: Try to add a console.log here to see if `this.jumpForwardInTimeOnCreation` has a proper value?

My hypothesis is that it's maybe too early to ask the renderer to update?

**Comment by tristanbob** [comment]
I got this working! 

I moved the function call from the constructor to update() and I added a boolean variable to function as "TriggerOnce".  

I have a few questions:

- Where is the best place to insert new functions/properties, etc? I have been trying to put them at the bottom of the namespace, except when something seemed to be at the bottom for a reason.
- Why did we need to define the properties twice in "ParticleSystem\particleemitterobject.ts" (once in `ParticleEmitterObjectDataType` and once in `ParticleEmitterObject`)?
- Was it correct to define my boolean `_jumpForwardInTimeCompleted` twice also?

Thanks for the help!

**Comment by tristanbob** [comment]
Does anyone have time to answer my questions on this PR?  (see my previous comment)

**Comment by 4ian** [comment]
> Where is the best place to insert new functions/properties, etc? I have been trying to put them at the bottom of the namespace, except when something seemed to be at the bottom for a reason.

Generally, adding after the rest is ok!

> Why did we need to define the properties twice in "ParticleSystem\particleemitterobject.ts" (once in ParticleEmitterObjectDataType and once in ParticleEmitterObject)?

`ParticleEmitterObjectDataType` describes what you're expecting to read from the project. 
`ParticleEmitterObject` is what lives in memory during the game. 
These are often similar, but not necessarily (typically, the boolean to know if something was done has no sense in the `ParticleEmitterObjectDataType`)


> Was it correct to define my boolean _jumpForwardInTimeCompleted twice also?

See my comment ;)

**Comment by 4ian** [comment]
Apart from one comment, I think this will be ready to merge!

File: Extensions/ParticleSystem/particleemitterobject.ts [hunk 2]
```diff
@@ -52,6 +52,8 @@ namespace gdjs {
     flow: number;
     /** Destroy the object when there is no particles? */
     destroyWhenNoParticles: boolean;
+    jumpForwardInTimeOnCreation: number;
+    _jumpForwardInTimeCompleted: boolean;
```
Comment (tristanbob) [inline Extensions/ParticleSystem/particleemitterobject.ts:56]:
```suggestion
```

**Comment by tristanbob** [comment]
> > Was it correct to define my boolean _jumpForwardInTimeCompleted twice also?
> 
> See my comment ;)

Thanks, I have removed the line you suggested.

(I accidentally made the change as a new comment instead of directly resolving your comment, but now I know how to do it on the next review).

BTW - I did a quick build with this commit and it worked as expected.

**Review by 4ian — APPROVED** [review]

**Pull request merged**